torch/nn/_functions/rnn.py

```diff
@@ -43,17 +46,18 @@ def LSTMCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
 
 
 def GRUCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
-
     if input.is_cuda:
-        gi = F.linear(input, w_ih)
+        gi = input.expand(3, input.size(0), input.size(1)).transpose(0, 1).contiguous() if w_ih is None else \
+            F.linear(input, w_ih)
         gh = F.linear(hidden, w_hh)
         state = fusedBackend.GRUFused()
         return state(gi, gh, hidden) if b_ih is None else state(gi, gh, hidden, b_ih, b_hh)
 
-    gi = F.linear(input, w_ih, b_ih)
+    gi = input.expand(3, input.size(0), input.size(1)).transpose(0, 1).contiguous() if w_ih is None else \
+        F.linear(input, w_ih, b_ih)
     gh = F.linear(hidden, w_hh, b_hh)
-    i_r, i_i, i_n = gi.chunk(3, 1)
-    h_r, h_i, h_n = gh.chunk(3, 1)
+    i_r, i_i, i_n = torch.unbind(gi.view(input.size(0), 3, -1), 1)
+    h_r, h_i, h_n = torch.unbind(gh.view(input.size(0), 3, -1), 1)
```

Comment:
Wasn't an issue for pre `skip_input` cos the output would go straight into the the next time cell state, but because of skip input we have to expand the tensor which ends up a different size (I can get this back to the normal shape, but that would involve doing for example `input.expand(4, input.size(0), input.size(1)).transpose(0, 1).contiguous().view(input.size(0), -1)` for the `x_h` variable in the LSTMCell!